codegen/smithy-dafny-codegen-cli/src/main/java/software/amazon/polymorph/CodegenCli.java

```diff
@@ -66,13 +80,54 @@ public static void main(String[] args) {
         cliArguments.outputDafnyDir.ifPresent(path -> outputDirs.put(TargetLanguage.DAFNY, path));
         cliArguments.outputJavaDir.ifPresent(path -> outputDirs.put(TargetLanguage.JAVA, path));
         cliArguments.outputDotnetDir.ifPresent(path -> outputDirs.put(TargetLanguage.DOTNET, path));
+        cliArguments.outputPythonDir.ifPresent(path -> outputDirs.put(TargetLanguage.PYTHON, path));
+
+        ObjectNode.Builder objectNodeBuilder = ObjectNode.builder();
+        // If a smithy-build.json is provided, parse it for settings to pipe into PluginContext
+        if (cliArguments.smithyBuildFilePath.isPresent()) {
+            String json;
+            try {
+                json = Files.readString(cliArguments.smithyBuildFilePath.get());
+            } catch (IOException e) {
+                throw new RuntimeException("IOException while reading smithy-build.json", e);
+            }
+
+            JsonFactory factory = new JsonFactory();
+            ObjectMapper mapper = new ObjectMapper(factory);
+            JsonNode rootNode;
+            try {
+                rootNode = mapper.readTree(json);
+            } catch (JsonProcessingException e) {
+                throw new RuntimeException("JsonProcessingException while processing smithy-build.json", e);
+            }
+
+            Iterator<Entry<String, JsonNode>> attributes = rootNode
+                // MUST be "plugins" per smithy-build.json spec
+                .findValue("plugins")
+                // MUST be same plugin name as the Dafny codegen plugin
+                .findValue(DafnyClientCodegenPlugin.pluginName).fields();
+
+            while(attributes.hasNext()) {
+                Entry<String, JsonNode> entry = attributes.next();
+                // Remove leading and trailing " on the attribute value before adding it to settings
+                objectNodeBuilder.withMember(entry.getKey(), entry.getValue().toString().replace("\"", ""));
+            }
+        }
+
+        final ServiceShape serviceShape = ModelUtils.serviceFromNamespace(serviceModel, cliArguments.namespace);
+        final PluginContext pluginContext = PluginContext.builder()
+            .model(serviceModel)
+            .fileManifest(FileManifest.create(cliArguments.outputPythonDir().orElse(cliArguments.modelPath)))
+            .settings(objectNodeBuilder.build())
+            .build();
```

Comment:
Question/Issue: 
I thought Robin and Alex had already implemented
a `smithy-build.json` parser.

It is probably not in the CLI project,
but the actual codegen project.

See: https://github.com/smithy-lang/smithy-dafny/tree/main-1.x/TestModels/aws-sdks/sqs

Reply:
I typed out a longer response, but I lost that somewhere in my tabs...
I believe they did not write a parser, but are instead using the standard Smithy `gradle` build process, which parses it for them.
We still need a way to pipe this information into the CLI.

Reply:
Oh, I would argue that we just should not.
Either we use the CLI without the JSON,
or we support the JSON via the gradle build process.

Otherwise, it feels like we are not adopting Smithy conventions but committing a work around to them...

But we can bring this up in a meeting.

Reply:
The CLI and the Smithy plugin both pipe their configuration through the common `CodegenEngine` class, so to support a new language you should only have to augment the set of CLI options and tweak the `DafnyClientCodegenPluginSettings` class (which is the deserialized model of the relevant sub-section of `smithy-build.json` - see `DafnyClientCodegenPluginSettings.fromObject`).

Reply:
This is updated. Check out https://github.com/lucasmcdonald3/smithy-dafny/pull/13